examples/MinioClientWithClientGrantsProvider.java

```diff
@@ -72,42 +68,41 @@ public static void main(String[] args) throws Exception {
         "https://IDP-HOST:IDP-PORT/auth/realms/master/protocol/openid-connect/token";
 
     // Client-ID to fetch JWT.
-    String clientId = "USER-ID";
+    String clientId = "IDP-CLIENT-ID";
 
     // Client secret to fetch JWT.
-    String clientSecret = "PASSWORD";
-
-    // Client-ID of MinIO service on IDP.
-    String idpClientId = "MINIO-CLIENT-ID";
+    String clientSecret = "IDP-CLIENT-SECRET";
 
     // STS endpoint usually point to MinIO server.
     String stsEndpoint = "http://STS-HOST:STS-PORT/";
 
-    // Policy to Credentials.
-    String policy =
-        new StringBuilder()
-            .append("{\n")
-            .append("    \"Statement\": [\n")
-            .append("       " + " {\n")
-            .append("            \"Action\": [\n")
-            .append("                \"s3:GetBucketLocation\",\n")
-            .append("                \"s3:ListBucket\"\n")
-            .append("            ],\n")
-            .append("            \"Effect\": \"Allow\",\n")
-            .append("            \"Principal\": \"*\",\n")
-            .append("            \"Resource\": \"arn:aws:s3:::test\"\n")
-            .append("        }\n")
-            .append("    ],\n")
-            .append("    \"Version\": \"2012-10-17\"\n")
-            .append("}\n")
-            .toString();
+    // Policy to Credentials. - optional, primary policy
+    // is inherited from JWT claims mapping.
+    //
+    // String policy =
+    //     new StringBuilder()
+    //         .append("{\n")
+    //         .append("    \"Statement\": [\n")
+    //         .append("       " + " {\n")
+    //         .append("            \"Action\": [\n")
+    //         .append("                \"s3:GetBucketLocation\",\n")
+    //         .append("                \"s3:ListBucket\"\n")
+    //         .append("            ],\n")
+    //         .append("            \"Effect\": \"Allow\",\n")
+    //         .append("            \"Principal\": \"*\",\n")
+    //         .append("            \"Resource\": \"arn:aws:s3:::test\"\n")
+    //         .append("        }\n")
+    //         .append("    ],\n")
+    //         .append("    \"Version\": \"2012-10-17\"\n")
+    //         .append("}\n")
+    //         .toString();
 
     Provider provider =
         new ClientGrantsProvider(
-            () -> getJwt(clientId, clientSecret, idpClientId, idpEndpoint),
+            () -> getJwt(clientId, clientSecret, idpEndpoint),
             stsEndpoint,
             null,
-            policy,
+            null, // policy, comment this line if you are adding additional policy restrictions
```

Comment:
You could have `String policy = null;` above line no.79 and no change here needed.